mysql/versions.go

```diff
@@ -0,0 +1,113 @@
+package mysql
+
+import (
+	"database/sql"
+	"github.com/hashicorp/go-version"
+	"strings"
+)
+
+type SQLVendor string
+
+const (
+	MySQL   SQLVendor = "MySQL"
+	MariaDB SQLVendor = "MariaDB"
+)
+
+type ServerVersion struct {
+	vendor  SQLVendor
+	version *version.Version
+}
+
+func serverVersion(db *sql.DB) (ServerVersion, error) {
+	versionString, err := serverVersionString(db)
+	if err != nil {
+		return ServerVersion{}, err
+	}
+
+	return parseServerVersionString(versionString), nil
+}
+
+func (serverVersion ServerVersion) supportsRoles() bool {
```

Comment:
Those helpers could receive `db *sql.DB` as well, that might simplify their usage.

What i like about this version is that it would enable a performance optimization: We could query for the version once and use that single result throughout all functions that need it.